#! /bin/sh /usr/share/dpatch/dpatch-run
## exitprompt.dpatch by Daniel Kristjansson <danielk@cuymedia.net>

@DPATCH@
diff -urNad mythtv-0.20+fixes15192~/libs/libmythtv/tv_play.h mythtv-0.20+fixes15192/libs/libmythtv/tv_play.h
--- blah/libs/libmythtv/tv_play.h	(revision 15199)
+++ blah/libs/libmythtv/tv_play.h	(working copy)
@@ -362,6 +362,7 @@
     QString baseFilters;
     QString db_time_format;
     QString db_short_date_format;
+    int     db_exit_prompt;
     int     fftime;
     int     rewtime;
     int     jumptime;
diff -urNad mythtv-0.20+fixes15192~/libs/libmythtv/tv_play.cpp mythtv-0.20+fixes15192/libs/libmythtv/tv_play.cpp
--- blah/libs/libmythtv/tv_play.cpp	(revision 15199)
+++ blah/libs/libmythtv/tv_play.cpp	(working copy)
@@ -267,6 +267,7 @@
     : QObject(NULL, "TV"),
       // Configuration variables from database
       baseFilters(""), db_time_format("h:mm AP"), db_short_date_format("M/d"),
+      db_exit_prompt(0),
       fftime(0), rewtime(0),
       jumptime(0), smartChannelChange(false),
       MuteIndividualChannels(false), arrowAccel(false),
@@ -382,6 +383,7 @@
     baseFilters         += gContext->GetSetting("CustomFilters");
     db_time_format       = gContext->GetSetting("TimeFormat", "h:mm AP");
     db_short_date_format = gContext->GetSetting("ShortDateFormat", "M/d");
+    db_exit_prompt       = gContext->GetNumSetting("PlaybackExitPrompt");
     db_use_picture_attr  = gContext->GetNumSetting(
                             "UseOutputPictureControls", 0);
     smartChannelChange   = gContext->GetNumSetting("SmartChannelChange", 0);
@@ -2487,8 +2489,7 @@
             NormalSpeed();
             StopFFRew();
 
-            if (StateIsPlaying(internalState) &&
-                gContext->GetNumSetting("PlaybackExitPrompt") == 1 && 
+            if (StateIsPlaying(internalState) && (1 == db_exit_prompt) &&
                 (!playbackinfo || !playbackinfo->isVideo)  )
             {
                 nvp->Pause();
@@ -2505,8 +2506,7 @@
                 if (GetOSD())
                     GetOSD()->NewDialogBox(dialogname, message, options, 0);
             }
-            else if (StateIsPlaying(internalState) &&
-                     gContext->GetNumSetting("PlaybackExitPrompt") == 1 && 
+            else if (StateIsPlaying(internalState) && (1 == db_exit_prompt) &&
                      playbackinfo && playbackinfo->isVideo && 
                      !activerbuffer->InDVDMenuOrStillFrame())
             {
@@ -2523,14 +2523,14 @@
             }
             else if (StateIsLiveTV(GetState()))
             {
-                if (nvp && gContext->GetNumSetting("PlaybackExitPrompt") == 2)
+                if (nvp && (2 == db_exit_prompt))
                     nvp->SetBookmark();
                 exitPlayer = true;
                 wantsToQuit = true;
             }
             else 
             {
-                if (nvp && gContext->GetNumSetting("PlaybackExitPrompt") == 2)
+                if (nvp && (2 == db_exit_prompt))
                     nvp->SetBookmark();
                 exitPlayer = true;
                 wantsToQuit = true;
@@ -5211,8 +5211,7 @@
         }
         else if (nvp && message.left(12) == "EXIT_TO_MENU")
         {
-            int exitprompt = gContext->GetNumSetting("PlaybackExitPrompt");
-            if (exitprompt == 1 || exitprompt == 2)
+            if (db_exit_prompt == 1 || db_exit_prompt == 2)
                 nvp->SetBookmark();
             wantsToQuit = true;
             exitPlayer = true;
